server/uno.go

```diff
@@ -222,34 +212,44 @@ func drawCard(gameID string, playerID string) (*model.Game, error) {
 		return nil, gameErr
 	}
 
-	// We loop through the players in the game
-	for _, player := range game.Players {
-		// We check that the current item has the same id as the one provided
-		if playerID == player.ID {
-			// Our player exists and we will talk a card from the draw pile
-			// and place it in the players hand
-
-			// We must make sure the draw pile is not empty. If empty move over discard pile,
-			// if discard pile is also empty... i have set it to add a new deck, probably should do something else.
-			if len(game.DrawPile) == 0 {
-				if len(game.DiscardPile) == 0 {
-					game.DrawPile = generateShuffledDeck()
-				} else {
-					game.DrawPile = shuffleCards(game.DiscardPile)
-					game.DiscardPile = game.DiscardPile[:0]
-				}
+	// We get the current player from the game
+	player := &game.Players[game.CurrentPlayer]
+	//We then check if the player attempting to play a card is the current player
+	if player.ID == playerID {
+		// We check if the draw pile has available cards
+		if len(game.DrawPile) == 0 {
+			// we check that the discard pile has cards to reshuffle
+			if len(game.DiscardPile) <= 1 {
```

Comment:
Nice work thinking of all these scenarios, like when there is nothing in the drawpile or discard pile